CMakeLists.txt

```diff
@@ -202,7 +202,21 @@ if("${BACKEND}" STREQUAL "HIP")
     list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} ${ROCM_PATH}/hip)
 
     # Set supported GPU Targets
-    set(DEFAULT_GPU_TARGETS "gfx908;gfx90a;gfx942;gfx1030;gfx1031;gfx1032;gfx1100;gfx1101;gfx1102;gfx1200;gfx1201")
+    if(NOT GPU_TARGETS AND NOT AMDGPU_TARGETS)
+        find_package(ROCmCMakeBuildTools QUIET)
+        if(NOT ROCmCMakeBuildTools_FOUND)
+            find_package(ROCM QUIET)
+        endif()
+        include(ROCMCheckTargetIds OPTIONAL RESULT_VARIABLE HAS_ROCM_CHECK_TARGET_IDS)
+
+        set(OPTIONAL_GPU_TARGETS "gfx942;gfx950;gfx1200;gfx1201")
```

Comment:
Why `gfx942 `is in the `OPTIONAL_GPU_TARGETS` list? Shouldn't it be in the `DEFAULT_GPU_TARGETS` list?

Reply:
The cutoff is arbitrary. I thought it was plausible that rpp would still work on a compiler old enough that gfx942 wasn't available, so I put it in the optional bucket. The only real tradeoff for putting targets in the optional bucket is that checking each target takes time. Feel free to move it if you wish.